subprojects/core/src/main/java/org/gradle/internal/cancel/DefaultBuildCancellationTokenFactory.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.cancel;
+
+import org.gradle.initialization.BuildCancellationToken;
+import org.gradle.initialization.DefaultBuildCancellationToken;
+
+public enum DefaultBuildCancellationTokenFactory implements BuildCancellationTokenFactory{
```

Comment:
Why does this class exist? It clearly doesn't do anything useful, because it just creates a new token every time, meaning that no one will actually be notified of any cancellation.

Reply:
In some cases we have to create an `ExecFactory` with a "false" cancellation token, for example in no-daemon environment, or [here](https://github.com/gradle/gradle/blob/2bafa9609490a29020eeeee4fe0c9a176afd8adc/subprojects/core/src/main/java/org/gradle/internal/service/scopes/BasicGlobalScopeServices.java#L87). In these cases, this class can be used to generate a token. Yes, this generated token doesn't do anything useful, but that's expected behavior.